NuGet.Docs/Contribute/Coding-Guidelines.md

```diff
@@ -1,102 +1,511 @@
+
+## Source code management
+
+:grey_exclamation: The *structure* of the code that we write and the *tools* that we use to write the code.
+
+### Repos
+
+To create a new repo in the https://github.com/nuget/ org, contact @yishaigalatzer.
+
+### Home repo
+
+The https://github.com/nuget/Home repo is the starting point for all things NuGet. It has the [issue tracker](https://github.com/nuget/home/issues) as well as an [up-to-date README](https://github.com/NuGet/Home/blob/master/README.md) of how repos are structured. Make sure to consult it before beginning your journey through NuGet code.
+
+### Branch strategy
+
+In general:
+
+* `master` has the code for the latest release of NuGet (e.g. alpha, beta, RC, RTM)
+* `dev` has the code that is being worked on but not yet released. This is the branch into which devs normally submit pull requests and merge changes into.
+* `release` has the code that is being staged and stabilized for an upcoming release
+
+Shortly before a release, the `release` branches are created from `dev`, and stabilization work happens there. Post-release work continus in the `dev` branch. Once the release takes place, the code is pushed from `release` to `master`.
+
+### Solution and project folder structure and naming
+
+Solution files generally go in the repo root.
+
+Solution names match repo names (e.g. NuGet.sln in the NuGet repo).
+
+Every project also needs a `project.json` and a matching `.kproj` file. This `project.json` is the source of truth for a project's dependencies and configuration options.
+
+Solutions need to contain solution folders that match the physical folders (`src`, `test`, etc.).
+
+For example, in the `Fruit` repo with the `Banana` and `Lychee` projects you would have these files checked in:
+
+```
+/Fruit.sln
+/src
+/src/NuGet.Banana
+/src/NuGet.Banana/project.json
+/src/NuGet.Banana/Banana.kproj
+/src/NuGet.Banana/Banana.cs
+/src/NuGet.Banana/Util/BananaUtil.cs
+/src/NuGet.Lychee
+/src/NuGet.Lychee/project.json
+/src/NuGet.Lychee/Lychee.kproj
+/src/NuGet.Lychee/Lychee.cs
+/src/NuGet.Lychee/Util/LycheeUtil.cs
+/test
+/test/NuGet.Banana.Tests
+/test/NuGet.Banana.Tests/project.json
+/test/NuGet.Banana.Tests/BananaTest.kproj
+/test/NuGet.Banana.Tests/BananaTest.cs
+/test/NuGet.Banana.Tests/Util/BananaUtilTest.cs
+```
+
+Note that after running the `build` command the system will generate the following files:
+
+```
+/build/*.shade
+```
+
+All these files are set to be ignored in the `.gitignore` file.
+
+### Conditional compilation for Desktop/CoreCLR
+
+Almost all development is done for both CoreCLR and Desktop .NET. Some code will be CoreCLR-specific or Desktop-specific because of API changes or behavior differences. The build system has two conditional compilation statements to assist with this:
+
+Desktop:
+
+```c#
+#ifdef ASPNET50
+```
+
+CoreCLR:
+
+```c#
+#ifdef ASPNETCORE50
```

Comment:
Fixed in https://github.com/NuGet/NuGetDocs/commit/81004c2142b59f7eb84617789fdffe153d330bb5. Needs an update in https://github.com/aspnet/Home/wiki/Engineering-guidelines as well.